includes/class-genesis-responsive-slider-admin.php

```diff
@@ -0,0 +1,432 @@
+<?php
+/**
+ * Genesis Responsive Slider Admin.
+ *
+ * @package genesis-responsive-slider
+ */
+
+/**
+ * Creates settings and outputs admin menu and settings page.
+ *
+ * @package genesis-responsive-slider
+ */
+class Genesis_Responsive_Slider_Admin {
+
+	/**
+	 * Constructor.
+	 */
+	public static function init() {
+		add_action( 'admin_init', array( 'Genesis_Responsive_Slider_Admin', 'register_genesis_responsive_slider_maybe_reset_settings' ) );
+		add_action( 'admin_notices', array( 'Genesis_Responsive_Slider_Admin', 'genesis_responsive_slider_notice' ) );
+		add_action( 'admin_menu', array( 'Genesis_Responsive_Slider_Admin', 'genesis_responsive_slider_settings_init' ), 15 );
+		add_filter( 'screen_layout_columns', array( 'Genesis_Responsive_Slider_Admin', 'genesis_responsive_slider_settings_layout_columns' ), 10, 2 );
+	}
+
+	/**
+	 * Reset settings.
+	 */
+	public static function register_genesis_responsive_slider_maybe_reset_settings() {
+
+		register_setting( GENESIS_RESPONSIVE_SLIDER_SETTINGS_FIELD, GENESIS_RESPONSIVE_SLIDER_SETTINGS_FIELD );
+		add_option( GENESIS_RESPONSIVE_SLIDER_SETTINGS_FIELD, Genesis_Responsive_Slider::genesis_responsive_slider_defaults(), '', 'yes' );
+
+		if ( Genesis_Responsive_Slider::genesis_get_responsive_slider_option( 'reset' ) ) {
+			update_option( GENESIS_RESPONSIVE_SLIDER_SETTINGS_FIELD, Genesis_Responsive_Slider::genesis_responsive_slider_defaults() );
+
+			genesis_admin_redirect( 'genesis_responsive_slider', array( 'reset' => 'true' ) );
+			exit;
+		}
+
+	}
+
+	/**
+	 * This is the notice that displays when you successfully save or reset
+	 * the slider settings.
+	 */
+	public static function genesis_responsive_slider_notice() {
+
```

Comment:
At the moment, these notices appear when saving any settings page, not just the slider settings page. For example:

1. Go to Settings -> Reading.
2. Click “Save Changes”.

You'll see the responsive slider “settings saved” notice.

The original function had this at the start to protect against this:

```php
if ( ! isset( $_REQUEST['page'] ) || $_REQUEST['page'] != 'genesis_responsive_slider' ) {
	return;
}
```

https://github.com/studiopress/genesis-responsive-slider/blob/17b0d83009d6b9f624c0da9e1fd8c39de2b9c68a/admin.php#L73-L74

We'll need to reinstate that here, or something similar. cc @marksabbath

Reply:
I've added it back @nickcernis :)